minlzlib/xzstream.c

```diff
@@ -369,7 +382,15 @@ XzDecodeBlockHeader (
     //
     if (!BfSeek(sizeof(*blockHeader), (uint8_t**)&blockHeader))
     {
-        return false;
+        return XzBlockHeaderFail;
+    }
+    if (blockHeader->Size == 0)
+    {
+        //
+        // That's no block! That's an index!
+        //
+        BfSeek((uint32_t)(-(uint16_t)sizeof(*blockHeader)), (uint8_t**)&blockHeader);
```

Comment:
1) Do we need the double casts or are we just being pedantic?
2) If they're really needed, is this going over 80 chars?

Reply:
I had to use both to get MSVC with `/W4 /WX` options set to accept the code. I welcome other ways to write this that gets past those options. (See https://godbolt.org/z/esEsGx)

We could use `1 + ~sizeof(*blockHeader)` , but that strikes me as less readable than negation, even with the casts.

Reply:
Lines rewrapped to 80 characters.